packages/tools/pluggable-widgets-tools/configs/rollup.config.js

```diff
@@ -0,0 +1,278 @@
+            // babel doesn't interfere with `import`s and `require`s used by rollup/commonjs plugin;
+            // also resulting code includes generated code that deserve transpilation as well.
+            getBabelInputPlugin({
+                sourceMaps: config.sourceMaps,
+                babelrc: false,
+                babelHelpers: "bundled",
+                plugins: ["@babel/plugin-proposal-class-properties"],
+                overrides: [
+                    {
+                        test: /node_modules/,
+                        plugins: ["@babel/plugin-transform-flow-strip-types", "@babel/plugin-transform-react-jsx"]
+                    },
+                    {
+                        exclude: /node_modules/,
+                        plugins: [["@babel/plugin-transform-react-jsx", { pragma: "createElement" }]]
+                    }
+                ]
+            }),
+            commonjs({ extensions: config.extensions, transformMixedEsModules: true, requireReturnsDefault: true }),
+            replace({
+                "process.env.NODE_ENV": production ? "'production'" : "'development'"
+            }),
+            config.transpile
+                ? getBabelOutputPlugin({
+                      sourceMaps: config.sourceMaps,
+                      babelrc: false,
+                      compact: false,
+                      ...(config.babelConfig || {})
+                  })
+                : null,
+            production ? terser() : null,
+            // We need to create .mpk and copy results to test project after bundling is finished.
+            // In case of a regular build is it is on `writeBundle` of the last config we define
+            // (since rollup processes configs sequentially). But in watch mode rollup re-bundles only
+            // configs affected by a change => we cannot know in advance which one will be "the last".
+            // So we run the same logic for all configs, letting the last one win.
+            command([
+                async () => {
+                    mkdirSync(mpkDir, { recursive: true });
+                    await zip(outDir, mpkFile);
+                    if (!production && projectPath) {
+                        cp("-r", join(outDir, "*"), join(projectPath, `deployment/${platform}/widgets`));
+                        cp(mpkFile, join(projectPath, "widgets"));
+                    }
+                }
+            ])
+        ];
+    }
+
+    function getClientComponentPlugins() {
+        return [
+            isTypescript ? widgetTyping({ sourceDir: join(sourcePath, "src") }) : null,
+            clear({ targets: [outDir, mpkDir] }),
+            command([() => cp(join(sourcePath, "src/**/*.xml"), outDir)]),
+            args.watch && platform === "web" && !production && projectPath ? livereload() : null
+        ];
+    }
+};
+
+function onwarn(warning, warn) {
+    if (["CIRCULAR_DEPENDENCY", "THIS_IS_UNDEFINED", "UNUSED_EXTERNAL_IMPORT"].includes(warning.code)) {
```

Comment:
I assume we are filtering the warnings UNUSED_EXTERNAL_IMPORT, THIS_IS_UNDEFINED ? why would that be useful ? 😬

Reply:
We basically white-list warnings, saying that these 3 are warnings while all others should be treated as errors. Unfortunately, there are some warnings from rollup that indicate quite severe issues, hence this code.

This 3 warnings i found to be unfixable. For example, `THIS_IS_UNDEFINED` is caused by react-jsxdev rewrite of functional components defined as a top-level arrow function.